compiler error with icc 17.0.4.196

I am writing to report another building problem of libfabric. Intel Compiler 17.0.4.196 Build 20170411 failed to build libfabric at the linking stage, regardless of the backend gcc versions of default 4.8.5 or 6.3.0 in the Red Hat Enterprise Linux Server release 7.3 system. Error messages follow,

```
../configure CC=icc CXX=icpc --enable-psm2 --disable-udp --disable-sockets --disable-rxm  --prefix=$NWCHEM_ROOT/deps


make -j 16 
<.....>
  CC       util/info.o
  CC       util/strerror.o
  CC       util/pingpong.o
  CCLD     libpsmx.la <http://libpsmx.la> 
  CCLD     libpsmx2.la <http://libpsmx2.la> 
  CCLD     src/libfabric.la <http://libfabric.la> 
  CCLD     util/fi_info
  CCLD     util/fi_strerror
  CCLD     util/fi_pingpong
src/.libs/libfabric.so: undefined reference to `__builtin_ia32_rdpmc'
src/.libs/libfabric.so: undefined reference to `__builtin_ia32_rdpmc'
src/.libs/libfabric.so: undefined reference to `__builtin_ia32_rdpmc'
make[1]: *** [util/fi_info] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [util/fi_strerror] Error 1
make[1]: *** [util/fi_pingpong] Error 1
make[1]: Leaving directory `/work1/home/u1/molpro/src/i17/nwchem-6.8.1.opt/libfabric/build'
make: *** [all] Error 2
```

--- a/configure.ac
+++ b/configure.ac
@@ -183,7 +183,12 @@ fi
 
 AC_CHECK_HEADER([linux/perf_event.h],
     [AC_CHECK_DECL([__builtin_ia32_rdpmc],
-        [linux_perf_rdpmc=1],
+        [
+            AC_TRY_LINK([#include <linux/perf_event.h>],
+                [__builtin_ia32_rdpmc(0);],
+                [linux_perf_rdpmc=1],
+                [linux_perf_rdpmc=0])
+	],
         [linux_perf_rdpmc=0],
         [#include <linux/perf_event.h>])],
     [linux_perf_rdpmc=0])

